scripts/cleanup/unusued_compilation.py

```diff
@@ -0,0 +1,129 @@
+#
+# Copyright (c) Contributors to the Open 3D Engine Project.
+# For complete copyright and license terms please see the LICENSE at the root of this distribution.
+#
+# SPDX-License-Identifier: Apache-2.0 OR MIT
+#
+#
+
+import argparse
+import fnmatch
+import os
+import shutil
+import sys
+sys.path.append(os.path.join(os.path.dirname(os.path.abspath(__file__)), '../build'))
+import ci_build
+
+EXTENSIONS_OF_INTEREST = ('.c', '.cc', '.cpp', '.cxx', '.h', '.hpp', '.hxx', '.inl')
```

Comment:
Is it worth adding .m / .mm files now in case this script ever gets used for mac/iOS?

Reply:
this script right now is brute-forcing on windows, so it wont file unused files in other platforms.
We can extend it to run in the local platform and include those files. But I am just running it in Windows because that is where the bulk of the code is.
I think we should take a different approach for platform-specific code, like just filtering in platform specific code since the rest of the code should have been analyzed by this pass.

Reply:
Happy to see this make it in! I am unclear if this is a worthwhile endeavor for Linux and Mac as those platforms are newer for O3DE and are less likely to have unused code.

Reply:
the script excludes anything that is platform specific (everything inside Platform folders) because e.g. traits can be deleted in some cases and will succeed.
There are a lot of things that this is likely missing. We could run it in other platforms and then try to find the intersection of things that are common, however, the amount of code we have per-platform is little in comparison, so even if we find a good percentage, the amount of removed lines is going to be quite low.
I would rather keep this route and start finding things inside files that may be unused, e.g. we could try to remove functions, or include headers.